components/places/src/db/schema.rs

```diff
@@ -325,6 +325,39 @@ pub fn upgrade_from(db: &Connection, from: u32) -> rusqlite::Result<()> {
                 rusqlite::params![NEW_SQL],
             )?;
         }
+        18 => {
+            // Fix any multi-level orphans by reparenting them to "unfiled_____". We'll do this in
+            // one shot using a recursive CTE.
+            db.execute_batch(
```

Comment:
@mhammond I was going to do this before the `ON CONFLICT` statement and log but rather not have this run on every sync (even if we do a one shot and add some kinda flag that we've done it). However I believe this would _prove_ the current theory that somehow the DB got into a bad state and dogear never ended fixing it up later. If true, we would start seeing the bookmark engine going back up and then the trigger would not end up breaking for migrated users moving forward

Reply:
I'm not sure what you mean. I'm very skeptical that 20% of the databases out there are corrupt, which is what this would imply given the ~20% of failures with the new check we backed out?

Reply:
shouldn't we do this only after we have evidence of a problem?

Reply:
You're right I think this is a little heavy-handed here. I moved this to the `apply_remote_items` part of the code since it'll help us give a better idea of the state of the users DB. I don't expect to fully keep the logging in there and this is only temporary.